### PR TITLE
Revert "Refactor temp_keys in preparation for dynamic captcha registration flow (#2618)"

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This reverts commit 60175a1b59795355a6054098c25a26e1b07efd4d.

The only changes kept from #2618 are two typo fixes in the comments. Reverting required to adapt the new flow module slightly.

### CODE REVIEW

File: src/internet_identity/src/state/temp_keys.rs [hunk 1]
```diff
@@ -121,17 +107,17 @@ impl TempKeys {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-struct TempKey {
+pub struct TempKey {
```
Comment (lmuntaner) [inline src/internet_identity/src/state/temp_keys.rs:110]: Is this still needed to be public?

File: src/internet_identity/src/state/temp_keys.rs [hunk 2]
```diff
@@ -121,17 +107,17 @@ impl TempKeys {
...
     /// The temp key principal
     principal: Principal,
     /// The expiration timestamp of the temp key
     expiration: Timestamp,
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-struct TempKeyExpiration {
+pub struct TempKeyExpiration {
```
Comment (lmuntaner) [inline src/internet_identity/src/state/temp_keys.rs:118]: ditto

**Review by lmuntaner — APPROVED** [review]
Thanks! Minor comments, already approving.

File: src/internet_identity/src/state/temp_keys.rs (hunk 1 above)
Comment (frederikrothenberger) [reply src/internet_identity/src/state/temp_keys.rs]: No, but it was like that before.
I removed it again.

**Pull request merged**